validate-artifacts.ps1

```diff
@@ -0,0 +1,39 @@
+
+# default script values 
+$artifactsConfigPath = "$PWD/artifacts.json"
+$resourcesTemplateName = "exported-template.json"
+$taskName = "task1"
+$tempFolderPath = "$PWD/temp"
+
+Write-Output "Reading config" 
+$artifactsConfig = Get-Content -Path $artifactsConfigPath | ConvertFrom-Json 
+
+Write-Output "Checking if temp folder exists"
+if (-not (Test-Path "$tempFolderPath")) { 
+    Write-Output "Temp folder does not exist, creating..."
+    New-Item -ItemType Directory -Path $tempFolderPath
+}
+
+Write-Output "Downloading artifacts"
+
+if (-not $artifactsConfig.resourcesTemplate) { 
+    Write-Error "Artifact config value 'resourcesTemplate' is empty! Please make sure that you executed the script 'scripts/generate-artifacts.ps1', and commited your changes"
+} 
+Invoke-WebRequest -Uri $artifactsConfig.resourcesTemplate -OutFile "$tempFolderPath/$resourcesTemplateName" -UseBasicParsing
+
+Write-Output "Validating artifacts"
+$TemplateFileText = [System.IO.File]::ReadAllText("$tempFolderPath/$resourcesTemplateName")
+$TemplateObject = ConvertFrom-Json $TemplateFileText -AsHashtable
+
+$storageAccount = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Storage/storageAccounts" )
```

Comment:
The script assumes that the JSON structure contains a `resources` array with objects having a `type` property. Ensure that the JSON file structure matches this expectation to avoid runtime errors.